### PR TITLE
Minor fixes to usages of RoutingStrategy and AddressTag

### PR DESCRIPTION
**Pull request by SzymonPobiega**
Allow fo custom routing strategies that modify the headers (the built-in ones don't do it).

### CODE REVIEW

**Comment by SzymonPobiega** [comment]
@Particular/nservicebus-maintainers please review.

**Comment by SimonCropp** [comment]
Looks ok. Is no test possible?

**Comment by SzymonPobiega** [comment]
@SimonCropp I've added a test for `RoutingToDispatchConnector`. The other behavior is currently broken (I have a fix for it waiting for #3641 to be merged) and can't be tested. The change in `TimeoutRecoverabilityBehavior` should be covered by ATs.

File: src/NServiceBus.Core.Tests/Routing/RoutingToDispatchConnectorTests.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    {
+        [Test]
+        public async Task It_preserves_headers_generated_by_custom_routing_strategy()
+        {
+            var behavior = new RoutingToDispatchConnector();
+            var message = new OutgoingMessage("ID", new Dictionary<string, string>(), new byte[0]);
+            await behavior.Invoke(new RoutingContext(message,
+                new CustomRoutingStrategy(), new FakeContext()), context =>
+                {
+                    Assert.IsTrue(context.Operations.First().Message.Headers.ContainsKey("CustomHeader"));
```
Comment (timbussmann) [inline src/NServiceBus.Core.Tests/Routing/RoutingToDispatchConnectorTests.cs]: I think we should have the assertions at the end of the test and not within a callback unless necessary.

Comment (SzymonPobiega) [inline src/NServiceBus.Core.Tests/Routing/RoutingToDispatchConnectorTests.cs]: Fixed.

**Comment by timbussmann** [comment]
Looks good to me

**Comment by SimonCropp** [comment]
@SzymonPobiega thanks for the clarification

**Pull request merged**